nodebuilder/fraud/service.go

```diff
@@ -2,21 +2,77 @@ package fraud
 
 import (
 	"context"
+	"encoding/json"
 
 	"github.com/celestiaorg/celestia-node/fraud"
 )
 
-// Module encompasses the behavior necessary to subscribe and broadcast fraud proofs within the network.
-// Any method signature changed here needs to also be changed in the API struct.
-type Module interface {
+var _ Module = (*Service)(nil)
+
+// Service is an implementation of Module that uses fraud.Service as a backend. It is used to provide fraud proofs as a
+// non-interface type to the API, and wrap fraud.Subscriber with a channel of Proofs.
+type Service struct {
 	fraud.Service
 }
 
-// API is a wrapper around Module for the RPC.
-// TODO(@distractedm1nd): These structs need to be autogenerated.
-//
-//go:generate go run github.com/golang/mock/mockgen -destination=mocks/api.go -package=mocks . Module
-type API struct {
-	Subscribe func(fraud.ProofType) (fraud.Subscription, error)
-	Get       func(context.Context, fraud.ProofType) ([]fraud.Proof, error)
+func (s *Service) Subscribe(ctx context.Context, proofType fraud.ProofType) (chan Proof, error) {
+	subscription, err := s.Service.Subscribe(proofType)
+	if err != nil {
+		return nil, err
+	}
+	concreteProofs := make(chan Proof)
+	go func() {
+		proof, _ := subscription.Proof(ctx)
+		concreteProofs <- Proof{Proof: proof}
```

Comment:
Why do we need a goroutine and channel here? Looks like it doesn't do any parallelisation.  If there is only one value to be obtained from subscription the routine that reads from channel will wait anyway, making obtaining proof a sequential operation.  Same result as would be without any goroutine/channel usage.

Reply:
we need a channel of a JSON-serializable type for the RPC, in order to do subscriptions. but take another look now, the first thing was just a prototype